instrumentation/github.com/emicklei/go-restful/otelrestful/internal/semconvutil/httpconv_test.go

```diff
@@ -165,47 +166,90 @@ func TestHTTPClientRequestRequired(t *testing.T) {
 }
 
 func TestHTTPServerRequest(t *testing.T) {
```

Comment:
The usual Go pattern for subtests like this is more like this:

```go
func TestHTTPServerRequest(t *testing.T) {
	for _, tt := range []struct {
		name                  string
		requestModifierFn     func(r *http.Request)
		httpServerRequestOpts HTTPServerRequestOptions

		wantClientIP string
	}{
		{
			name:         "with a client IP from the network",
			wantClientIP: "1.2.3.4",
		},
		{
			name: "with a client IP from x-forwarded-for header",
			requestModifierFn: func(r *http.Request) {
				r.Header.Add("X-Forwarded-For", "5.6.7.8")
			},
			wantClientIP: "5.6.7.8",
		},
		{
			name: "with a cl;ient IP in options",
			requestModifierFn: func(r *http.Request) {
				r.Header.Add("X-Forwarded-For", "5.6.7.8")
			},
			httpServerRequestOpts: HTTPServerRequestOptions{
				HTTPClientIP: "9.8.7.6",
			},
			wantClientIP: "9.8.7.6",
		},
	} {
		t.Run(tt.name, func(t *testing.T) {
			reqCh := make(chan *http.Request, 1)
			handler := func(w http.ResponseWriter, r *http.Request) {
				r.RemoteAddr = "1.2.3.4:5678"
				reqCh <- r
				w.WriteHeader(http.StatusOK)
			}

			srv := httptest.NewServer(http.HandlerFunc(handler))
			defer srv.Close()

			srvURL, err := url.Parse(srv.URL)
			require.NoError(t, err)
			srvPort, err := strconv.ParseInt(srvURL.Port(), 10, 32)
			require.NoError(t, err)

			req, err := http.NewRequest(http.MethodGet, srv.URL, nil)
			require.NoError(t, err)

			if tt.requestModifierFn != nil {
				tt.requestModifierFn(req)
			}

			resp, err := srv.Client().Do(req)
			require.NoError(t, err)
			require.NoError(t, resp.Body.Close())

			var got *http.Request
			select {
			case got = <-reqCh:
				// All good
			case <-time.After(5 * time.Second):
				t.Fatal("Did not receive a signal in 5s")
			}

			peer, peerPort := splitHostPort(got.RemoteAddr)

			const user = "alice"
			got.SetBasicAuth(user, "pswrd")

			assert.ElementsMatch(t,
				[]attribute.KeyValue{
					attribute.String("http.method", "GET"),
					attribute.String("http.scheme", "http"),
					attribute.String("net.host.name", srvURL.Hostname()),
					attribute.Int("net.host.port", int(srvPort)),
					attribute.String("net.sock.peer.addr", peer),
					attribute.Int("net.sock.peer.port", peerPort),
					attribute.String("user_agent.original", "Go-http-client/1.1"),
					attribute.String("http.client_ip", tt.wantClientIP),
					attribute.String("net.protocol.version", "1.1"),
					attribute.String("http.target", "/"),
				},
				HTTPServerRequest("", got, tt.httpServerRequestOpts))
		})
	}
}
```

Which forces each subtest to follow the same pattern.

Reply:
Changed